integration-tests/ahm/src/tests.rs

```diff
@@ -344,3 +349,115 @@ async fn migration_works() {
 
 	println!("Migration done in {} RC blocks", rc_block_count);
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn scheduled_migration_works() {
+	let Some((mut rc, mut ah)) = load_externalities().await else { return };
+
+	// Check that the migration is pending on the RC.
+	rc.execute_with(|| {
+		log::info!("Asserting the initial state on RC");
+		next_block_rc();
+
+		assert_eq!(RcMigrationStageStorage::<Polkadot>::get(), RcMigrationStage::Pending);
+
+		// clear the DMP queue.
+		let _ = DownwardMessageQueues::<Polkadot>::take(AH_PARA_ID);
+	});
+	rc.commit_all().unwrap();
+
+	// Check that the migration is pending on the AH.
+	ah.execute_with(|| {
+		log::info!("Asserting the initial state on AH");
+		next_block_ah();
+
+		assert_eq!(AhMigrationStageStorage::<AssetHub>::get(), AhMigrationStage::Pending);
+
+		// clear the UMP queue.
+		let _ = PendingUpwardMessages::<AssetHub>::take();
+	});
+	ah.commit_all().unwrap();
+
+	// Schedule the migration on RC.
+	let dmp_messages = rc.execute_with(|| {
+		log::info!("Scheduling the migration on RC");
+		next_block_rc();
+
+		let now = frame_system::Pallet::<Polkadot>::block_number();
+		let scheduled_at = now + 2;
```

Comment:
The `schedule_migration` should probably check that the scheduled block number is inn the future